sktime/performance_metrics/forecasting/_classes.py

```diff
@@ -1,54 +1,1304 @@
+        self.square_root = square_root
+        super().__init__(func=func, name=name, greater_is_better=greater_is_better)
+
+
+class _SquaredPercentageMetricFunctionWrapper(
+    _SquaredPercentageErrorMixIn, _MetricFunctionWrapper
+):
+    def __init__(
+        self,
+        func,
+        name=None,
+        greater_is_better=False,
+        square_root=False,
+        symmetric=True,
+    ):
+        self.square_root = square_root
+        self.symmetric = symmetric
+        super().__init__(func=func, name=name, greater_is_better=greater_is_better)
+
+
+class _AsymmetricMetricFunctionWrapper(_AsymmetricErrorMixIn, _MetricFunctionWrapper):
+    def __init__(
+        self,
+        func,
+        name=None,
+        greater_is_better=False,
+        asymmetric_threshold=0,
+        left_error_function="squared",
+        right_error_function="absolute",
+    ):
+        self.asymmetric_threshold = asymmetric_threshold
+        self.left_error_function = left_error_function
+        self.right_error_function = right_error_function
+        super().__init__(func=func, name=name, greater_is_better=greater_is_better)
+
+
+class _RelativeLossMetricFunctionWrapper(_RelativeLossMixIn, _MetricFunctionWrapper):
+    def __init__(
+        self,
+        func,
+        name=None,
+        greater_is_better=False,
+        relative_loss_function=mean_absolute_error,
+    ):
+        self.relative_loss_function = relative_loss_function
+        super().__init__(func=func, name=name, greater_is_better=greater_is_better)
+
+
+def make_forecasting_scorer(func, name=None, greater_is_better=False):
+    """Factory method for creating metric classes from metric functions
+
+    Parameters
+    ----------
+    func:
+        Loss function to convert to a forecasting scorer class
+
+    name: str, default=None
+        Name to use for the forecasting scorer loss class
+
+    greater_is_better: bool, default=False
+        If True then maximizing the metric is better.
+        If False then minimizing the metric is better.
+
+    Returns
+    -------
+    scorer:
+        Metric class that can be used as forecasting scorer.
+    """
+
+    return _MetricFunctionWrapper(func, name=name, greater_is_better=greater_is_better)
+
+
+class MeanAbsoluteScaledError(_ScaledMetricFunctionWrapper):
+    """Mean absolute scaled error (MASE). MASE output is non-negative floating
+    point. The best value is 0.0.
+
+    This scale-free error metric can be used to compare forecast methods on
+    a single series and also to compare forecast accuracy between series.
+
+    This metric is well suited to intermittent-demand series because it
+    will not give infinite or undefined values unless the training data
+    is a flat timeseries. In this case the function returns a large value
+    instead of inf.
+
+    Works with multioutput (multivariate) timeseries data
+    with homogeneous seasonal periodicity.
+
+    Parameters
+    ----------
+    sp : int
```

Comment:
why does this `sp` parameter appear in docstrings, like here and elsewhere? It does not seem to be a hyper-parameter (settable in `__init__`)

Reply:
(understanding that it's an attribute set by some mixins)